File: app/src/main/java/pl/kwako/metroid_map/view/ImageCoordinateTranslator.java
package pl.kwako.metroid_map.view;

import pl.kwako.metroid_map.Settings;

import javax.inject.Inject;

class ImageCoordinateTranslator {

    private final Settings settings;

    @Inject
    public ImageCoordinateTranslator(Settings settings) {
        this.settings = settings;
    }

    public int toImageX(int x) {
        return (x * settings.roomImgWidth());
    }

    public int toImageY(int y) {
        return (y * settings.roomImgHeight());
    }
}

File: app/src/main/kotlin/pl/kwako/metroid_map/view/ImageCoordinateTranslator.kt
package pl.kwako.metroid_map.view

import pl.kwako.metroid_map.Settings

import javax.inject.Inject

class ImageCoordinateTranslator @Inject constructor(private val settings: Settings) {

    fun toImageX(x: Int): Int {
        return x * settings.roomImgWidth
    }

    fun toImageY(y: Int): Int {
        return y * settings.roomImgHeight
    }
}
